src/ansys/fluent/core/services/interceptors.py

```diff
@@ -94,6 +94,31 @@ def intercept_unary_unary(
         return self._intercept_call(continuation, client_call_details, request)
 
 
+class GrpcErrorInterceptor(grpc.UnaryUnaryClientInterceptor):
+    """Interceptor class to check Fluent server error state before gRPC calls are
+    made."""
+
+    def _intercept_call(
+        self,
+        continuation: Any,
+        client_call_details: grpc.ClientCallDetails,
+        request: Any,
+    ) -> Any:
+        response = continuation(client_call_details, request)
+        if response.exception() is not None and response.code() != grpc.StatusCode.OK:
+            raise RuntimeError(response.details(), response.code())
```

Comment:
I think it is better to display only error details string instead of a tuple. With the following code you can get the original behaviour (won't require to change the unitests):
```
grpc_ex = response.exception()
ex = RuntimeError(grpc_ex.details())
ex.__context__ = grpc_ex
raise ex from None
```

Reply:
Thank you for this suggestion @mkundu1. Implemented the same.